### PR TITLE
PT-1303 add new fields to PA

### PR DESCRIPTION
**Pull request by franck-fourel**
- Add eventType and eventID
- inputTime will be added later once validated as part of https://github.com/mdblp/platform/pull/51

### CODE REVIEW

File: data/types/activity/physical/physical.go [hunk 1]
```diff
@@ -196,6 +205,9 @@ type Physical struct {
 	Name              *string          `json:"name,omitempty" bson:"name,omitempty"`
 	ReportedIntensity *string          `json:"reportedIntensity,omitempty" bson:"reportedIntensity,omitempty"`
 	Step              *Step            `json:"step,omitempty" bson:"step,omitempty"`
+	EventID           *string          `json:"eventId,omitempty" bson:"eventId,omitempty"`
+	EventType         *string          `json:"eventType,omitempty" bson:"eventType,omitempty"`
+	// inputTime
```
Comment (yvon-dblg) [inline data/types/activity/physical/physical.go]: The inputTime field is documented in commit 694e5ac1ff57b4b8d5d6e684702cf495df38d239
Why it is not implemented ?
If not needed anymore/for now, a more explicit comment is needed.

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
I have one comment/question, plus this remark not fully related to this PR:
I didn't check the PR for the spec, but all examples provided the timezoneOffset only, not the timezone.
The offset can be known from the timezone, the opposite is not true. In fact, the offset is not really needed.
We should provide example with the timezone information to not breaks what we have done on this point before.

File: data/types/activity/physical/physical.go (hunk 1 above)
Comment (franck-fourel) [reply data/types/activity/physical/physical.go]: inputTime is included as common object in #51 
That's the reason why it's not there. Waiting for approval of #51 to include it.

File: data/types/activity/physical/physical.go [hunk 2]
```diff
@@ -265,6 +274,11 @@ func (p *Physical) Validate(validator structure.Validator) {
 	if p.Step != nil {
 		p.Step.Validate(validator.WithReference("step"))
 	}
+	p.EventType.Validate(validator)
```
Comment (yvon-dblg) [inline data/types/activity/physical/physical.go]: The documentation should mention this behavior:
eventId is mandatory when the eventType is specified.

File: data/types/activity/physical/physical.go [hunk 3]
```diff
@@ -265,6 +274,11 @@ func (p *Physical) Validate(validator structure.Validator) {
...
+	if p.EventType.EventType != nil {
+		validator.String("eventId", p.EventID).Exists()
```
Comment (yvon-dblg) [inline data/types/activity/physical/physical.go]: The duration should be mandatory in this case

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
2 questions, one I think must be change.

File: data/types/activity/physical/physical.go (hunk 2 above)
Comment (franck-fourel) [reply data/types/activity/physical/physical.go]: that's correct

File: data/types/activity/physical/physical.go (hunk 3 above)
Comment (franck-fourel) [reply data/types/activity/physical/physical.go]: makes sense

**Review by yvon-dblg — APPROVED** [review]
LGTM

File: docs/UpdatedData.md [hunk 4]
```diff
@@ -179,13 +179,10 @@ __Note #2__: the `"part":2` object is not mandatory. The system can decide to ca
...
+- `eventId`: unique ID provided by the client that is used to link stop and start events. Iw we receive several objects with the same ID, the most recent one will be the effective one while the other objects will be considered as history of changes. The duration is __mandatory__ when this field is provided.
 
-In the below example, the physical activity is entered on the handset at 8:00am. It starts at 8:50am for 60 minutes. Finally the stop event says that activity stopped at 9:40, that is tha actual duration was 50 minutes. This last information was entered at 10:00am.
+In the below example, The 2 objects are sent in 2 separate reauests. The first object coming with the first request indicates that the physical activity is entered on the handset at 8:00am. It starts at 8:50am for 60 minutes. The second object that is received later on as part of a second request says that the duration of the same activity has been changed to 50 minutes. This last information was entered at 10:00am. This second object will become the effctive one while first one can be considered as the history of changes. The link between both is done through the common `eventId`.
```
Comment (CedricPerotto) [inline docs/UpdatedData.md]:
```suggestion
In the below example, The 2 objects are sent in 2 separate requests. The first object coming with the first request indicates that the physical activity is entered on the handset at 8:00am. It starts at 8:50am for 60 minutes. The second object that is received later on as part of a second request says that the duration of the same activity has been changed to 50 minutes. This last information was entered at 10:00am. This second object will become the effective one while first one can be considered as the history of changes. The link between both is done through the common `eventId`. 
```

File: docs/UpdatedData.md [hunk 5]
```diff
@@ -179,13 +179,10 @@ __Note #2__: the `"part":2` object is not mandatory. The system can decide to ca
...
-- `inputTime` is a UTC string timestamp that defines at what time the patient has entered the physical activity. This field is optional. It takes the same format as `time` field.
-- `eventType`: type of event, either `start` or `end`
-  - `start` defines the beginning of the event. The `duration` is the estimated one. The `time` field gives the actual start time of the event.
-  - `stop` gives the end of the event. The `duration` is the actual duration. The `time` field gives the actual end time of the event. 
-- `eventId`: unique ID provided by the client that is used to link stop and start events.
+- `inputTime` is a UTC string timestamp that defines at what time the patient has entered or modified the physical activity. This field is optional. It takes the same format as `time` field.
+- `eventId`: unique ID provided by the client that is used to link stop and start events. Iw we receive several objects with the same ID, the most recent one will be the effective one while the other objects will be considered as history of changes. The duration is __mandatory__ when this field is provided.
```
Comment (CedricPerotto) [inline docs/UpdatedData.md]:
```suggestion
- `eventId`: unique ID provided by the client that is used to link stop and start events. If we receive several objects with the same ID, the most recent one will be the effective one while the other objects will be considered as history of changes. The duration is __mandatory__ when this field is provided.
```

**Review by CedricPerotto — CHANGES_REQUESTED** [review]
LGTM, just typos in the doc.
@yvon-dblg ?

File: data/types/common/eventtype.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package common
```
Comment (yvon-dblg) [inline data/types/common/eventtype.go]: Do we keep theses files ?

**Review by yvon-dblg — COMMENTED** [review]
Still working :)

File: data/types/common/eventtype.go (hunk 6 above)
Comment (franck-fourel) [reply data/types/common/eventtype.go]: deleted as part of f6df691

**Comment by franck-fourel** [comment]
@yvon-dblg  ready for final review

**Review by yvon-dblg — APPROVED** [review]
LGTM
@CedricPerotto ?

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**